File: Java/02- Relacionando Classes/16- Contrato/Fisica.java

package com.mycompany.mavenproject6;

public abstract class Fisica extends Pessoa{
    protected Sexo sexo;
    protected  EstadoCivil estadocivil;
    protected String dataNascimento;

    public Fisica(Sexo sexo, EstadoCivil estadocivil, String dataNascimento, String nome, String telefone, String email, Endereco endereco) {
        super(nome, telefone, email, endereco);
        this.sexo = sexo;
        this.estadocivil = estadocivil;
        this.dataNascimento = dataNascimento;
    }

    public Sexo getSexo() {
        return sexo;
    }

    public void setSexo(Sexo sexo) {
        this.sexo = sexo;
    }

    public EstadoCivil getEstadocivil() {
        return estadocivil;
    }

    public void setEstadocivil(EstadoCivil estadocivil) {
        this.estadocivil = estadocivil;
    }

    public String getDataNascimento() {
        return dataNascimento;
    }

    public void setDataNascimento(String dataNascimento) {
        this.dataNascimento = dataNascimento;
    }

  @Override
    public String toString() {
        return super.toString() +
               "\n=====Dados da Pessoa Fisica=====" + 
               "\nSexo: " + sexo.getTexto() + " | " + sexo.getCaractere() +
               "\nData Nascimento: " + dataNascimento;
    }
    
}

File: Python/04- POO/03- Atividade Pontuada/models/fisica.py
from models.pessoa import Pessoa
from models.enum.sexo import Sexo


class Fisica(Pessoa):
    def __init__(self, nome, telefone, email, endereco, cpf: str, rg: str, data_nascimento: str, sexo: Sexo) -> None:
        super().__init__(nome, telefone, email, endereco)
        self.cpf = cpf
        self.rg = rg
        self.data_nascimento = data_nascimento
        self.sexo = sexo

    def __str__(self) -> str:
        return (f"{super().__str__()}" 
                f"\nCPF: {self.cpf}"
                f"\nRG: {self.rg}"
                f"\nData de Nascimento: {self.data_nascimento}"
                f"\nSexo: {self.sexo.texto} | {self.sexo.caractere}"
                )